iot_verifier/src/poc.rs

```diff
@@ -508,6 +513,31 @@ fn verify_witness_distance(
     Ok(())
 }
 
+/// verify min hex distance between beaconer and witness
+fn verify_witness_cell_distance(
+    beacon_loc: Option<u64>,
+    witness_loc: Option<u64>,
+) -> GenericVerifyResult {
+    // other verifications handle location checks but dont assume
+    // we have a valid location passed in here
+    // if no location for either beaconer or witness then default
+    // this verification to a fail
+    let l1 = beacon_loc.ok_or(InvalidReason::BelowMinDistance)?;
+    let l2 = witness_loc.ok_or(InvalidReason::BelowMinDistance)?;
+    let cell_distance = match calc_cell_distance(l1, l2) {
+        Ok(d) => d,
+        Err(_) => return Err(InvalidReason::BelowMinDistance),
+    };
+    if cell_distance < POC_CELL_DISTANCE_MINIMUM {
+        tracing::debug!(
+            "witness verification failed, reason: {:?}. cell distance {cell_distance}",
+            InvalidReason::BelowMinDistance
+        );
+        return Err(InvalidReason::BelowMinDistance);
+    }
```

Comment:
```suggestion
    if let (Some(beacon_loc), Some(witness_loc)) = (beacon_loc, witness_loc) {
        calc_cell_distance(beacon_loc, witness_loc).map_err(|_| InvalidReason::BelowMinDistance)?;
        return Ok(())
    }
    tracing::debug!("witness verification failed, reason: {:?}. cell distance {cell_distance}", InvalidReason::BelowMinDistance);
    Err(InvalidReason::BelowMinDistance)
```

Reply:
untested but could simplify the double-option situation you're looking to validate where both locations are "Some"

Reply:
i think i prefer the existing patten, more readable but will give it a go locally

Reply:
thanks for suggesting but i will keep as is for time being.